tools/toolchain/gen_mksurf_namelist.py

```diff
@@ -0,0 +1,658 @@
+                    const = True,
+                    required = False,
+                    dest="crop_flag", 
+                    default=True)
+        parser.add_argument('-f','--fast', 
+                    help='Toggle fast mode which does not user the large mapping file', 
+                    action="store_true", 
+                    dest="fast_flag", 
+                    default=False)
+        parser.add_argument('-r','--res',
+                    help='''
+                    Resolution is the supported resolution(s) to use for files.
+                    [default: %(default)s]
+                    ''',
+                    action="store", 
+                    dest="res", 
+                    choices=valid_opts['res'], 
+                    required=False, 
+                    default="4x5")
+        return parser
+
+
+def str2bool(v):
+    """
+    Function for converting different forms of
+    command line boolean strings to boolean value.
+
+    Args:
+        v (str): String bool input
+
+    Raises:
+        if the argument is not an acceptable boolean string
+        (such as yes or no ; true or false ; y or n ; t or f ; 0 or 1).
+        argparse.ArgumentTypeError: The string should be one of the mentioned values.
+
+    Returns:
+        bool: Boolean value corresponding to the input.
+    """
+    if isinstance(v, bool):
+       return v
+    if v.lower() in ('yes', 'true', 't', 'y', '1'):
+        return True
+    elif v.lower() in ('no', 'false', 'f', 'n', '0'):
+        return False
+    else:
+        raise argparse.ArgumentTypeError('Boolean value expected. [true or false] or [y or n]')
+
+def tag_describe ():
+    """
+    Function for converting different forms of
+    command line boolean strings to boolean value.
+
+    Args:
+
+    Raises:
+
+    Returns:
+        label.decode (str) : ouput of running 'git describe' in shell
+    """
+    label = subprocess.check_output(["git", "describe"]).strip()
+    return label.decode()
+
+def glc_nec_type(x):
+    """
+    Function for defining acceptable glc_nec input.
+
+    Args:
+        x (str) : glc_nec value from command line args.
+
+    Raises:
+        Error if value of glc_nec is not in the range 
+        of 1-99. 
+
+    Returns:
+        x (int) : Acceptable glc_nec value.
+    """
+    x = int(x)
+    if (x <= 0) or (x >= 100):
+        raise argparse.ArgumentTypeError("ERROR: glc_nec must be between 1 and 99.")
+    return x
+
+def start_year_type(x):
+    """
+    Function for defining acceptable start_year input.
+
+    Args:
+        x (str) : start_year string from command line args.
+
+    Raises:
+        Error if value of glc_start_year is not in the range 
+        of 850-2015. 
+
+    Returns:
+        x (int) : Acceptable start_year value.
+    """
+    x = int(x)
+    if (x < 850) or (x > 2105):
+        raise argparse.ArgumentTypeError(
+                "ERROR: Simulation start year should be between 850 and 2105.")
+    return x
```

Comment:
Just a thought (this is nit-picky of me :-) )

Could the last two "_type" functions become one function where the range values and part of the error msg come in as arguments? Only if you feel like doing it.

Reply:
I am not sure what you mean here. There are two functions here for (defining types) for command line arguments one is `glc_nec` and one is for `start_year`. I don't think we can combine them and use them in the parser and I don't see a point for doing that. This is how the code shows errors and warnings where glc_nec or year is not valid. I am marking this as resolved.